t_ja5/test_ja5t.py

```diff
@@ -0,0 +1,250 @@
+DURATION = int(tf_cfg.cfg.get("General", "duration"))
+
+
+class TestJa5tBase(tester.TempestaTest):
+    tempesta = {
+        "config": """
+listen 443 proto=h2,https;
+
+server ${server_ip}:8000;
+
+frang_limits {http_strict_host_checking false;}
+
+tls_certificate ${tempesta_workdir}/tempesta.crt;
+tls_certificate_key ${tempesta_workdir}/tempesta.key;
+tls_match_any_server_name;
+""",
+    }
+
+    backends = [DEPROXY_SERVER]
+
+    def setup_test(self):
+        self.start_all_services()
+
+        srv: StaticDeproxyServer = self.get_server("deproxy")
+        srv.set_response(
+            "HTTP/1.1 200 OK\r\n"
+            + "Connection: keep-alive\r\n"
+            + "Content-Length: 13\r\n"
+            + "Content-Type: text/html\r\n"
+            + "\r\n"
+            + "<html></html>"
+        )
+
+
+@marks.parameterize_class(
+    [
+        {"name": "Http", "clients": [DEPROXY_CLIENT_SSL]},
+        {"name": "H2", "clients": [DEPROXY_CLIENT_H2]},
+    ]
+)
+class TestJa5t(TestJa5tBase):
+    """This class contains checks for tempesta ja5 filtration."""
+
+    @marks.Parameterize.expand(
+        [
+            marks.Param(name="no_hash", tempesta_ja5_config="", expected_response=True),
+            marks.Param(
+                name="hash_not_for_client",
+                tempesta_ja5_config="""
+					ja5t {
+						hash deadbeef 10 1000;
+					}
+				""",
+                expected_response=True,
+            ),
+            marks.Param(
+                name="hash_for_client_not_block",
+                tempesta_ja5_config="""
+					ja5t {
+						hash 66cb9fd8d4250000 1 10;
+					}
+				""",
+                expected_response=True,
+            ),
+            marks.Param(
+                name="hash_for_client_block_by_conn",
+                tempesta_ja5_config="""
+					ja5t {
+						hash 66cb9fd8d4250000 0 10;
+						hash 66cb8f00d4250002 0 10;
+					}
+				""",
+                expected_response=False,
+            ),
+            marks.Param(
+                name="hash_for_client_block_by_rate",
+                tempesta_ja5_config="""
+					ja5t {
+						hash 66cb9fd8d4250000 1 0;
+						hash 66cb8f00d4250002 1 0;
+					}
+				""",
+                expected_response=False,
+            ),
+        ]
+    )
+    def test(self, name, tempesta_ja5_config: str, expected_response: bool):
+        """Update tempesta config. Send many identical requests and checks cache operation."""
+        tempesta: Tempesta = self.get_tempesta()
+        tempesta.config.defconfig += tempesta_ja5_config
+
+        self.setup_test()
+        client = self.get_client("deproxy")
+        request = client.create_request(method="GET", uri="/", headers=[])
+
+        client.make_request(request)
+        if expected_response:
+            self.assertTrue(client.wait_for_response())
+            self.assertTrue(client.last_response.status, "200")
+        else:
+            self.assertFalse(client.wait_for_response())
```

Comment:
Can we use `expected_block` here? It is unnecessary timeout...
```suggestion
        if expected_block:
            self.assertTrue(client.wait_for_connection_close())
        else:
            self.assertTrue(client.wait_for_response())
            self.assertTrue(client.last_response.status, "200")
```